File: logger-rate-limiter/logger-rate-limiter.cpp
class Logger {
    map<string,int>mp;
public:
    Logger() {
        
    }
    
    bool shouldPrintMessage(int timestamp, string message) {
        if(mp[message]>timestamp)
            return false;
        mp[message]=10+timestamp;
        return true;
    }
};

/**
 * Your Logger object will be instantiated and called as such:
 * Logger* obj = new Logger();
 * bool param_1 = obj->shouldPrintMessage(timestamp,message);
 */

File: 359-logger-rate-limiter/logger-rate-limiter.py
class Logger:

    def __init__(self):
        self.mp = defaultdict(str)

    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:
        if message in self.mp:
            if (timestamp - self.mp[message]) >= 10:
                self.mp[message] = timestamp
                return True
            else:
                return False
        else:
            self.mp[message] = timestamp
            return True


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)